crates/iota-transaction-builder/src/lib.rs

```diff
@@ -1,7 +1,277 @@
+//! [with_client](TransactionBuilder::with_client) method, the resulting builder
+//! will use it to find and validate provided IDs.
+//!
+//! ### Example with Client Resolution
+//!
+//! ```
+//! # use std::str::FromStr;
+//! use iota_graphql_client::Client;
+//! use iota_transaction_builder::TransactionBuilder;
+//! use iota_types::{Address, ObjectId, Transaction};
+//!
+//! # #[tokio::main(flavor = "current_thread")]
+//! # async fn main() -> eyre::Result<()> {
+//!
+//! let sender =
+//!     Address::from_str("0x611830d3641a68f94a690dcc25d1f4b0dac948325ac18f6dd32564371735f32c")?;
+//! let to_address =
+//!     Address::from_str("0x0000a4984bd495d4346fa208ddff4f5d5e5ad48c21dec631ddebc99809f16900")?;
+//!
+//! let mut builder = TransactionBuilder::new(sender).with_client(Client::new_devnet());
+//!
+//! let coin =
+//!     ObjectId::from_str("0x8ef4259fa2a3499826fa4b8aebeb1d8e478cf5397d05361c96438940b43d28c9")?;
+//!
+//! builder.send_coins([coin], to_address, 50000000000u64);
+//!
+//! let txn: Transaction = builder.finish().await?;
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! ### Example without Client Resolution
+//!
+//! ```
+//! # use std::str::FromStr;
+//! use iota_transaction_builder::TransactionBuilder;
+//! use iota_types::{Address, Digest, ObjectId, ObjectReference, Transaction};
+//!
+//! let sender =
+//!     Address::from_str("0x611830d3641a68f94a690dcc25d1f4b0dac948325ac18f6dd32564371735f32c")?;
+//! let to_address =
+//!     Address::from_str("0x0000a4984bd495d4346fa208ddff4f5d5e5ad48c21dec631ddebc99809f16900")?;
+//!
+//! let mut builder = TransactionBuilder::new(sender);
+//!
+//! let coin = ObjectReference {
+//!     object_id: ObjectId::from_str(
+//!         "0x8ef4259fa2a3499826fa4b8aebeb1d8e478cf5397d05361c96438940b43d28c9",
+//!     )?,
+//!     digest: Digest::from_str("4jJMQScR4z5kK3vchvDEFYTiCkZPEYdvttpi3iTj1gEW")?,
+//!     version: 435090179,
+//! };
+//! let gas_coin = ObjectReference {
+//!     object_id: ObjectId::from_str(
+//!         "0xd04077fe3b6fad13b3d4ed0d535b7ca92afcac8f0f2a0e0925fb9f4f0b30c699",
+//!     )?,
+//!     digest: Digest::from_str("8ahH5RXFnK1jttQEWTypYX7MRzLuQDEXk7fhMHCyZekX")?,
+//!     version: 473053810,
+//! };
+//!
+//! builder
+//!     .send_coins([coin], to_address, 50000000000u64)
+//!     .gas(gas_coin)
+//!     .gas_budget(1000000000)
+//!     .gas_price(100);
+//!
+//! let txn: Transaction = builder.finish()?;
+//! # Result::<_, eyre::Error>::Ok(())
+//! ```
+//!
+//! NOTE: It is possible to provide an [ObjectId](iota_types::ObjectId) to an
+//! offline client builder, but this will cause the builder to fail when calling
+//! `finish`.
+//!
+//! ## Methods
+//!
+//! There are three kinds of methods available:
+//!
+//! ### Commands
+//!
+//! Each command method adds one or more commands to the final transaction. Some
+//! commands have optional follow-up methods. All command results can be named
+//! via [name](TransactionBuilder::name);
```

Comment:
I would mention why naming a command is necessary because it's not apparent why that's a good feature to have?